### PR TITLE
SONARPY-1448 Rule S6730: Deprecated NumPy aliases of built-in types should not be used

### PR DESCRIPTION
**Pull request by joke1196**

### CODE REVIEW

File: python-checks/src/main/java/org/sonar/python/checks/DeprecatedNumpyTypesCheck.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.quickfix.TextEditUtils;
+
+@Rule(key = "S6730")
+public class DeprecatedNumpyTypesCheck extends PythonSubscriptionCheck {
+
+  private static final String MESSAGE = "Replace this deprecated numpy type alias with the builtin type %s.";
+  private static final String QUICK_FIX_MESSAGE = "Replace with %s.";
+
+  private static final Map<String, String> typesToCheck = Map.of(
```
Comment (maksim-grebeniuk-sonarsource) [inline python-checks/src/main/java/org/sonar/python/checks/DeprecatedNumpyTypesCheck.java:42]: I guess it is good to use a constant naming strategy here like `TYPES_TO_CHECK`

File: python-checks/src/main/java/org/sonar/python/checks/DeprecatedNumpyTypesCheck.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        DeprecatedNumpyTypesCheck::checkForDeprecatedTypesNames);
+  }
+
+  private static void checkForDeprecatedTypesNames(SubscriptionContext ctx) {
+    QualifiedExpression expression = (QualifiedExpression) ctx.syntaxNode();
+    Symbol symbol = expression.symbol();
+    checkForNumpyType(symbol, expression, ctx);
+  }
+
+  private static void checkForNumpyType(@Nullable Symbol symbol, Tree expression, SubscriptionContext ctx) {
```
Comment (maksim-grebeniuk-sonarsource) [inline python-checks/src/main/java/org/sonar/python/checks/DeprecatedNumpyTypesCheck.java:64]: Minor: I think there is not that much sense to have the separate method here. I guess the whole logic could be done in the `checkForDeprecatedTypeNames` method like:
```java
Optional.of(ctx.syntaxNode)
  .map(QualifiedExpression.class::cast)
  .ifPresent(expression -> Optional.ofNullable(expression.symbol)
    .map(Symbol::fullyQualifiedName)
    .map(typesToCheck::get)
    .ifPresent(type -> raiseIssue(expression, type, ctx)));
```

**Review by maksim-grebeniuk-sonarsource — APPROVED** [review]
LGTM, see a couple of minor hints, feel free to ignore

File: python-checks/src/main/java/org/sonar/python/checks/DeprecatedNumpyTypesCheck.java (hunk 2 above)
Comment (joke1196) [reply python-checks/src/main/java/org/sonar/python/checks/DeprecatedNumpyTypesCheck.java:64]: Yes, good catch! Thank you. This was a left over from some old implementation.

**Pull request merged**